modules/discretedata/dataset.h

```diff
@@ -0,0 +1,209 @@
+
+    /** Returns the specified channel, returns first instance found
+    *   @param name Unique name of requested channel
+    *   @param definedOn GridPrimitive type the channel is defined on, default 0D vertices
+    */
+    SharedConstChannel getChannel(const std::string& name,
+                                  GridPrimitive definedOn = GridPrimitive::Vertex) const;
+
+    /** Returns the specified channel if it is in the desired format, returns first instance found
+    *   @param name Unique name of requested channel
+    *   @param definedOn GridPrimitive type the channel is defined on, default 0D vertices
+    */
+    template <typename T>
+    std::shared_ptr<const DataChannel<T>> getChannel(
+        const std::string& name, GridPrimitive definedOn = GridPrimitive::Vertex) const;
+
+    /** Returns the specified channel if it is in the desired format, returns first instance found
+    *   @param name Unique name of requested channel
+    *   @param definedOn GridPrimitive type the channel is defined on, default 0D vertices
+    */
+    template <typename T>
+    std::shared_ptr<const DataChannel<T>> getChannel(
+        const char* name, GridPrimitive definedOn = GridPrimitive::Vertex) const;
+
+    /** Returns the specified buffer, converts to buffer or copies
+    *   @param name Unique name of requested buffer
+    *   @param definedOn GridPrimitive type the channel is defined on, default 0D vertices
+    */
+    template <typename T>
+    std::shared_ptr<const BufferChannel<T>> getAsBuffer(
+        const std::string& name, GridPrimitive definedOn = GridPrimitive::Vertex) const;
+
+    /** Remove channel from set by shared pointer, data remains valid if shared outside
+    *   Swaps position in vector
+    *   @param channel Shared pointer to data
+    *   @return Successfull - channel was saved in the set indeed
+    */
+    bool removeChannel(SharedConstChannel channel);
+
+    /** Number of channels currently held
+    */
+    ind numChannels() const { return ChannelSet.size(); }
+
+    // Attributes
+public:
+    /** Set of data channels
+    *   Indexed by name and defining dimension (0D vertices, 1D edges etc).
+    */
+    BaseChannelMap ChannelSet;
+};
+
+/** \class DataSet
+    \brief Data package containing structure by cell connectivity and data
+
+    Conglomerate of data grid and several data channels assigned to grid dimensions.
+
+
+    @author Anke Friederici and Tino Weinkauf
+*/
+class IVW_MODULE_DISCRETEDATA_API DataSet {
+
+    // Construction / Deconstruction
+public:
+    DataSet() = default;
+    virtual ~DataSet() = default;
+
+    /** Default copy shares Channels and Connectivity */
+    DataSet(const DataSet& copy) = default;
+
+    // Methods
+public:
+    /// Returns a const typed shared pointer to the grid, if casting is possible.
+    template <typename T>
+    std::shared_ptr<const T> getGrid() const {
+        return std::dynamic_pointer_cast<const T, const Connectivity>(Grid);
```

Comment:
you don't need the second template arg to dynamic_pointer_cast it is deduced

Reply:
Removed. 414bc9f9dd1e522163234430364c04fae58d4a48